src/ScriptCs.Core/ScriptLibraryComposer.cs

```diff
@@ -0,0 +1,117 @@
+
+            _fileSystem = fileSystem;
+            _preProcessor = preProcessor;
+            _packageContainer = packageContainer;
+            _packageAssemblyResolver = packageAssemblyResolver;
+            _logger = logger;
+        }
+
+        internal string GetMainScript(IPackageObject package)
+        {
+            var script =
+                package.GetContentFiles()
+                    .SingleOrDefault(f => f.EndsWith("main.csx", StringComparison.InvariantCultureIgnoreCase));
+            
+            if (script != null)
+            {
+                _logger.DebugFormat("Found main script: {0}", script);    
+            }
+            
+            return script;
+        }
+
+        public virtual string ScriptLibrariesFile
+        {
+            get { return "ScriptLibraries.csx"; }
+        }
+
+        public void Compose(StringBuilder builder = null)
+        {
+            var namespaces = new List<string>();
+            var references = new List<string>();
+
+            var packagesPath = Path.Combine(_fileSystem.CurrentDirectory, _fileSystem.PackagesFolder);
+            var packageReferences = _packageAssemblyResolver.GetPackages(_fileSystem.CurrentDirectory);
+            var packageScriptsPath = Path.Combine(packagesPath, ScriptLibrariesFile);
+
+            if (!_fileSystem.DirectoryExists(packagesPath) || _fileSystem.FileExists(packageScriptsPath))
+            {
+                return;
+            }
+
+            if (builder == null)
+            {
+                builder = new StringBuilder();
+            }
+
+            foreach (var reference in packageReferences)
+            {
+                ProcessPackage(packagesPath, reference, builder, references, namespaces);
+            }
+
+            foreach (var ns in namespaces)
+            {
+                builder.Insert(0, String.Format("using {0};{1}", ns, Environment.NewLine));
+            }
+
+            foreach (var reference in references)
+            {
+                builder.Insert(0, String.Format("#r {0}{1}", reference, Environment.NewLine));
+            }
+            _fileSystem.WriteToFile(packageScriptsPath, builder.ToString());
+        }
+
+        protected internal virtual void ProcessPackage(string packagesPath, IPackageReference reference, StringBuilder builder, List<string> references,
+            List<string> namespaces)
+        {
+            _logger.DebugFormat("Finding package:{0}", reference.PackageId);
+            var package = _packageContainer.FindPackage(packagesPath, reference);
+            _logger.Debug("Package found");
+            var script = GetMainScript(package);
+            if (script != null)
+            {
+                script = Path.Combine(packagesPath, string.Format("{0}.{1}", package.Id, package.TextVersion), script);
+                _logger.Debug("Pre-processing script");
+                var result = _preProcessor.ProcessFile(script);
```

Comment:
In the acceptance tests CalculatorMain.cs has a #load of Multiply.csx and that works fine. Where is the gap? Why do I care about LoadedScripts, is this to set the env or something?